### PR TITLE
cupy.sparse reference

### PR DESCRIPTION
**Pull request by unnonouno**
I made a reference page of cupy.sparse.
relate to #36

### CODE REVIEW

File: docs/source/reference/sparse.rst [hunk 1]
```diff
@@ -0,0 +1,15 @@
+-------------
+Sparse matrix
+-------------
+
+CuPy supports sparse matrix using `cuSPARSE <https://developer.nvidia.com/cusparse>`_.
```
Comment (takagi) [inline docs/source/reference/sparse.rst]: "sparse matrices" would be better to be followed by the next line.

File: docs/source/reference/sparse.rst [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+These matrices have the same interfaces of `SciPy's sparse matrices <https://docs.scipy.org/doc/scipy/reference/sparse.html>`_.
+
+.. autosummary::
+   :toctree: generated/
+   :nosignatures:
+
+   cupy.sparse.coo_matrix
```
Comment (takagi) [inline docs/source/reference/sparse.rst]: sphinx complains as following:
```
/home/mtakagi/chainer-test/cupy/docs/source/reference/sparse.rst:8: WARNING: failed to import cupy.sparse.coo_matrix
/home/mtakagi/chainer-test/cupy/docs/source/reference/sparse.rst:8: WARNING: failed to import cupy.sparse.csr_matrix
/home/mtakagi/chainer-test/cupy/docs/source/reference/sparse.rst:8: WARNING: failed to import cupy.sparse.csc_matrix
/home/mtakagi/chainer-test/cupy/docs/source/reference/sparse.rst:8: WARNING: toctree references unknown document u'reference/generated/cupy.sparse.coo_matrix'
/home/mtakagi/chainer-test/cupy/docs/source/reference/sparse.rst:8: WARNING: toctree references unknown document u'reference/generated/cupy.sparse.csr_matrix'
/home/mtakagi/chainer-test/cupy/docs/source/reference/sparse.rst:8: WARNING: toctree references unknown document u'reference/generated/cupy.sparse.csc_matrix'
```

**Comment by unnonouno** [comment]
I update it.

**Comment by takagi** [comment]
Do you have further update for the reference? Can I merge after checking?

**Comment by takagi** [comment]
LGTM!

**Pull request merged**